tdms/tests/unit/test_fields.cpp

```diff
@@ -0,0 +1,57 @@
+#include "catch2/catch_test_macros.hpp"
+#include "complex"
+#include "field.h"
+
+using namespace std;
+
+
+inline bool is_close(double a, double b){
+  return abs(a - b) < 1E-10;
+}
+
+inline bool is_close(complex<double> a, complex<double> b){
+  return is_close(a.real(), b.real()) && is_close(a.imag(), b.imag());
```

Comment:
Also, Julia's `isapprox` always compares `abs(a - b)`, which would be slightly different than operating on real and imaginary parts separately as you're doing here, the difference is comparing a rectangle vs a circle, but this is probably less important than adjusting the scale.

Reply:
not sure if comparing a circle is more or less intuitive to be honest, but will adjust. shouldn't change anything